### ISSUE TITLE
Fix creating tinyint or smallint for Avro in Hive

### ISSUE THREAD

**Issue by electrum** [prestodb/presto#6373]
Avro only supports `int`. In Hive, creating a table with `tinyint` or `smallint` with Avro storage format creates a table with `int` automatically. In Presto, we currently create a broken table (reading it fails with an Avro schema mismatch).

We should either disallow it as not supported (telling the user to use `integer` instead), or automatically translate the types.

**Comment by wagnermarkd** [comment]
This is an issue with how the Avro SerDe manages metadata and is also broken in Hive: https://issues.apache.org/jira/browse/HIVE-14509.

**Comment by ebyhr** [comment]
It seems that this is already fixed by #6370 :)
```
presto:test1> 
presto:test1> CREATE TABLE hive.web.page_views3 ( user_id tinyint ) WITH ( format = 'AVRO' );
Query 20170207_055948_00013_tnr8u failed: Column user_id is tinyint, which is not supported by Avro. Use integer instead.

presto:test1> CREATE TABLE hive.web.page_views3 ( user_id smallint ) WITH ( format = 'AVRO' );
Query 20170207_055957_00014_tnr8u failed: Column user_id is smallint, which is not supported by Avro. Use integer instead.

presto:test1> 
presto:test1> 
```
